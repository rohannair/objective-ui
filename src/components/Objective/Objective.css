@import '../../styles/globals';

.Objective {
  display: flex;
  flex-direction: column;
  width: 100%;
  padding: 0 20px;
}

.label {
  color: $secondary;
  margin-bottom: 0;
}

.valueContainer {
  font-size: 0.85rem;
  margin: 0 0 15px;
}

.buttonContainer {
  display: flex;
  flex-direction: row;
  width: 100%;
  align-items: center;
}

/* WTF */
