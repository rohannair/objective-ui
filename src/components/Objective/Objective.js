import React, { PropTypes } from 'react'
import styles from './Objective.css'

import Button from '../Button'

const Objective = p => {
  if (p.data) {
    return (
      <div className={styles.Objective}>
        <h4 className={styles.label}>Objective:</h4>
        <p className={styles.valueContainer}>{ p.data.name }</p>

        <h4 className={styles.label}>Key Results:</h4>
        <ul className={styles.valueContainer}>
          { p.data.key_results.map(v => <li key={`${p.id}-${v.id}`}>{v.name}</li>)}
        </ul>

        <h4 className={styles.label}>Timeline:</h4>
        <p className={styles.valueContainer}>{ p.data.timeline }</p>

        <div className={styles.actionBar}>
          <div className={styles.action} onClick={e => {
            e.preventDefault()
            p.editAction()
          }}><i className="zmdi zmdi-edit" /></div>
        </div>
      </div>
    )
  }

  return (
    <div className={styles.ObjectiveButton}>
      <Button primary onClick={p.buttonAction}>{p.buttonText}</Button>
    </div>
  )

}

export default Objective
