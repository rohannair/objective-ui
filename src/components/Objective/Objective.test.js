import React from 'react'
import Objective from './index.js'

describe('Test suite for Objective component', () => {
  it('Objective should exist', () => {
    let wrapper = shallow(<Objective/>)
    expect(wrapper).toMatchSnapshot()
  })
})

