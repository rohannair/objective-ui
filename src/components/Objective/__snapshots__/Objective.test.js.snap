exports[`Test suite for Objective component Objective should exist 1`] = `
<div>
  <Button
    primary={true} />
</div>
`;
